src/Type/ObjectType/CommentAuthor.php

```diff
@@ -56,8 +54,10 @@ public static function register_type() {
 
 						],
 						'resolve' => function ( $comment_author, $args, $context, $info ) {
+							// If the user cant see the CommentAuthor's email, grab it from the comment.
```

Comment:
We should update this comment to provide more information on why we're taking this approach. 

The intent of the Model layer is to keep private data from being used where it shouldn't be used. 

This bug is a result of the Model layer doing its job correctly, and I think we should clarify in this comment for users and our future selves why we're bypassing the Model layer and using a native function to get the email address instead. 

We've determined the `User->email` is a private field for public users. By default, we don't want public users to arbitrarily make queries and get user emails. 

This bug is a result of that. 

Of course, in this case, getting the email in the resolver and returning something (an avatar) should be considered "safe" because it doesn't expose the users email address publicly, instead it's transformed into an Avatar.

I want users to understand why we're bypassing the model layer here, so that anyone that comes across this code understands it's an exception, not the rule. 

It's still "best practice" to centralize permission checks in the Model layer. This is an exception to that best practice.

Reply:
How's this?
```php
/**
 * The model only returns emails for authenticated users, but we need a valid email to return an avatar for unauthenticated requests.
 * If the email isn't visible, we use the comment ID to retrieve it.
 */
 ```

Reply:
I think it's a shame that you need to go to the database again for the email that the Model already contains. A typical use case is listing the last comments. This will lead to an unnecessary increase in the number of queries to the database for an email that we already know and is only "hidden". It makes sense to take email from the database only for commenters who commented as not logged in. In this case, the email is assigned to every single comment and not to the user.

If we make email available in the model in a field that is not defined in Type, then there is no risk of public exposure. I add this note to reflect on efficiency.

Reply:
@romcik according to the [WP developer documentation](https://developer.wordpress.org/reference/functions/get_comment/), the comment should already be cached, so this shouldn't cause an extra database hit. 

If I'm wrong, please feel free to open up a new issue to address this, as this PR was already merged _and released_ so it'll probably get lost and forgotten about if the discussion stays here 🙏